extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java

```diff
@@ -372,10 +378,17 @@ private void visitAnnotation(final DotName clazz, final IndexView index, final M
                         CDI_PRODUCES_ANNOTATION,
                         target,
                         Collections.emptyList()));
-                annotationsToAdd.add(create(
-                        CDI_DEPENDENT_ANNOTATION,
-                        target,
-                        Collections.emptyList()));
+                DotName declaredScope = getScope(methodInfo);
+                // Non-spring specific annotations are processed by Arc
+                if (declaredScope == null && !methodInfo.hasAnnotation(CDI_SINGLETON_ANNOTATION)) {
```

Comment:
Can you clarify the second check a bit?

Reply:
It's linked to the previous review comment about the null check.